src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5547,13 +5607,14 @@ void markMembersAsNotNull(int receiverSlot, TypeSymbol type, string memberName,
         }
 
         private ImmutableArray<VisitArgumentResult> VisitArgumentsEvaluate(
-            SyntaxNode syntax,
             ImmutableArray<BoundExpression> arguments,
             ImmutableArray<RefKind> refKindsOpt,
             ImmutableArray<ParameterSymbol> parametersOpt,
             ImmutableArray<int> argsToParamsOpt,
             BitVector defaultArguments,
-            bool expanded)
+            bool expanded,
+            BoundExpression? receiver,
+            NullableFlowState receiverState)
```

Comment:
>receiverState [](http://example.com/codeflow?start=30&length=13)

It looks like we always pass one of the two values:
- default
- NotNull.

More over, default is passed only when we explicitly pass ```nulll``` for the receiver. What is the advantage of using ```default``` state vs. NotNull state in cases like that? Perhaps we don't need to pass the ```receiverState``` explicitly at all?